sahi/model.py

```diff
@@ -1063,3 +1063,106 @@ def _create_object_prediction_list_from_original_predictions(
             object_prediction_list_per_image.append(object_prediction_list)
 
         self._object_prediction_list_per_image = object_prediction_list_per_image
+
+
+@check_requirements(["tensorflow", "tensorflow_hub"])
+class TensorflowHubDetectionModel(DetectionModel):
+    def load_model(self):
+        try:
+            import tensorflow_hub as hub
+
+        except ImportError:
+            raise ImportError(
+                "Please install tensorflow and tensorflow_hub via `pip install tensorflow' and 'pip install tensorflow_hub`"
+            )
+
+        self.model = hub.load(self.model_path)
+
+    def perform_inference(self, image: np.ndarray):
+        if self.image_size is not None:
+            from sahi.utils.tfhub import get_image, resize
+
+            img = get_image(image)
+            img = resize(img, self.image_size)
+
+            prediction_result = self.model(img)
+
+        else:
+            from sahi.utils.tfhub import get_image
+
+            img = get_image(image)
+
+            prediction_result = self.model(img)
+
+        self._original_predictions = prediction_result
+        self.img = get_image(image)
+
+        if self.category_mapping is None:
+            from sahi.utils.tfhub import COCO_CLASSES
+
+            category_mapping = {str(i): COCO_CLASSES[i] for i in range(len(COCO_CLASSES))}
+            self.category_mapping = category_mapping
+
+    @property
+    def num_categories(self):
+        num_categories = len(self.category_mapping)
+        return num_categories
+
+    @property
+    def has_mask(self):
+        return self.model.with_mask
```

Comment:
tensorflow modellerinde `with_mask` diye bir property oldugundan emin misin?

Reply:
Açıkcası with_mask olayını çok anlamadım. Yolov5/detectron/trochvision de böyle bir fonksiyon olup olmadığını kontrol ettim bulamadım. O yüzden aynı şekilde yazdım.

Reply:
`with_mask` su anlama geliyor: model cikti olarak instance segmentation maskesi uretiyor mu.

yolov5'teki hatali olmus bir pr'la kaldirayim onu. yolov5'te maske olma ihtimali yok.

mmdet'te instance segmentation modellerinin maskesi oluyor.

torchvision ve detectronda instance segmentation modellerini desteklemiyoruz, onlarda da kaldirabiliriz o parametreyi.

simdilik tfhub icin de instance segmentation model destegini sonraya birakabiliriz `# TODO:` seklinde comment ekleyerek.

Reply:
Detectron2 de destekliyor diye hatırlıyorum. Eğer problem varsa tfhub dan sonra torchvision ve detectron2 için PR açabilrim. Tfhub için TODO ekliyorum.

Reply:
tamamdir bu pr'dam sonra digerlerinin maske ciktilarini isleme uzerine odaklanabiliriz.